src/yb/yql/pgwrapper/pg_libpq-test.cc

```diff
@@ -190,6 +190,36 @@ TEST_F(PgLibPqTest, Simple) {
   ASSERT_EQ(row, (decltype(row){1, "hello"}));
 }
 
+TEST_F(PgLibPqTest, PgStatIdxScanNoIncrementOnErrorTest) {
+  auto conn = ASSERT_RESULT(Connect());
+
+  std::string createTableStmt = "create table many (";
+  for (int i = 1; i <= 30; ++i)
+    createTableStmt += Format("c$0 int,", i);
+  createTableStmt += "k int primary key)";
+
+  std::string createIndexStmt = "create index on many (";
+  for (int i = 1; i <= 30; ++i)
+    createIndexStmt += Format("c$0,", i);
+  createIndexStmt += "c1 asc)";
+
+  std::string predicateErrorQuery = "select * from many where ";
+  for (int i = 1; i <= 30; ++i)
+    predicateErrorQuery += Format("c$0 = 1 and c$0 < 2 and c$0 > 0 and ", i);
+  predicateErrorQuery += " true";
```

Comment:
should try to find the lowest number of predicates that still hits error instead of this current 90.  I think it is 65.  Try storing `constexpr kMaxPredicates = 64` and using `kMaxPredicates + 1`.